app/controllers/HowYouWillGetYourExpensesController.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import javax.inject.Inject
+import models.TaxYearSelection
+import models.TaxYearSelection.{CurrentYear, CurrentYearMinus1}
+import pages.authenticated.TaxYearSelectionPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.{HowYouWillGetYourExpensesCurrentAndPreviousYearView, HowYouWillGetYourExpensesCurrentView, HowYouWillGetYourExpensesPreviousView}
+import scala.concurrent.ExecutionContext
+
+class HowYouWillGetYourExpensesController @Inject()(
+                                       override val messagesApi: MessagesApi,
+                                       identify: AuthenticatedIdentifierAction,
+                                       getData: DataRetrievalAction,
+                                       requireData: DataRequiredAction,
+                                       val controllerComponents: MessagesControllerComponents,
+                                       currentView: HowYouWillGetYourExpensesCurrentView,
+                                       previousView: HowYouWillGetYourExpensesPreviousView,
+                                       currentAndPreviousYearView: HowYouWillGetYourExpensesCurrentAndPreviousYearView
+                                     )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+
+      request.userAnswers.get(TaxYearSelectionPage) match {
+        case Some(x) if x.contains(CurrentYear) && x.length>1 => Ok(currentAndPreviousYearView("", containsCurrentYearMinus1(x)))
+        case Some(x) if x.contains(CurrentYear) => Ok(currentView(""))
+        case Some(x) => Ok(previousView("",containsCurrentYearMinus1(x)))
```

Comment:
Can we change 'x' to TaxYearSelection please? Just so it reads a little bit better.